enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,542 @@
+Example below shows annotations of a Pod with two `bridge` CNI interfaces:
+```yaml
+k8s.v1.cni.cncf.io/network-status:
+                    [{
+                        "name": "ovn-kubernetes",
+                        "interface": "eth0",
+                        "ips": [
+                            "10.42.0.18"
+                        ],
+                        "mac": "0a:58:0a:2a:00:12",
+                        "default": true,
+                        "dns": {}
+                    },{
+                        "name": "default/bridge-conf",
+                        "interface": "net1",
+                        "ips": [
+                            "10.10.1.20"
+                        ],
+                        "mac": "6e:4a:f5:49:fb:6d",
+                        "dns": {}
+                    },{
+                        "name": "default/bridge-conf",
+                        "interface": "net2",
+                        "ips": [
+                            "10.10.1.21"
+                        ],
+                        "mac": "02:4b:86:bf:80:1f",
+                        "dns": {}
+                    }]
+                  k8s.v1.cni.cncf.io/networks: bridge-conf,bridge-conf
+```
+
+There is an annotation for NetworkAttachmentDefinition - `k8s.v1.cni.cncf.io/resourceName`.
+It is used when some CNI requires information about specific device that is prepared by device plugin.
+It is primarily used with SR-IOV and as such out of scope for this enhancement mentioned only for completeness.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Not applicable
+
+#### Standalone Clusters
+
+Not applicable
+
+#### Single-node Deployments or MicroShift
+
+Enhancement is solely intended for MicroShift.
+
+### Implementation Details/Notes/Constraints [optional]
+
+First, it must be noted that Multus CNI itself is a meta-CNI. From a high level perspective, its
+purpose is to call other CNIs according to the CNI configs supplied by the user in form of
+NetworkAttachmentDefinitions and Pod annotations.
+Any specific actions are related to the delegate CNIs. For example: creating veth pair,
+attaching one end of veth to the bridge and making the other end available within the Pod is
+responsibility of `bridge` CNI is this example.
+MicroShift team will create tests for CNIs that will be declared as supported. However these tests
+will not explore the breadth and depth of possible network setups, so ultimately the responsibility
+for correctness of the configuration is up to the user.
+
+#### Manifests
+
+MicroShift will provide Multus based on thin architecture because of the resource consumption
+(see alternatives for more information).
+Because of the differences with OpenShift (which uses thick Multus plugin), existing OpenShift
+manifests will require changes to make them suitable for MicroShift. These manifests will reside
+in MicroShift repository.
```

Comment:
It's like OLM - we provide RPM with manifests and there's no way to customize them further at a later stage. If someone wants to change the behavior then they can fork, change it and rebuild.

Something like this would require support for Helm in addition to kustomize, or provide some microshift specific Helm-like some mechanism to customize the kustomize manifests